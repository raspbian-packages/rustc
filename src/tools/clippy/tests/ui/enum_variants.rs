#![feature(plugin, non_ascii_idents)]

#![warn(clippy, pub_enum_variant_names)]

enum FakeCallType {
    CALL, CREATE
}

enum FakeCallType2 {
    CALL, CREATELL
}

enum Foo {
    cFoo,
    cBar,
    cBaz,
}

enum Fooo {
    cFoo, // no error, threshold is 3 variants by default
    cBar,
}

enum Food {
    FoodGood,
    FoodMiddle,
    FoodBad,
}

enum Stuff {
    StuffBad, // no error
}

enum BadCallType {
    CallTypeCall,
    CallTypeCreate,
    CallTypeDestroy,
}

enum TwoCallType { // no error
    CallTypeCall,
    CallTypeCreate,
}

enum Consts {
    ConstantInt,
    ConstantCake,
    ConstantLie,
}

enum Two { // no error here
    ConstantInt,
    ConstantInfer,
}

enum Something {
    CCall,
    CCreate,
    CCryogenize,
}

enum Seal {
    With,
    Without,
}

enum Seall {
    With,
    WithOut,
    Withbroken,
}

enum Sealll {
    With,
    WithOut,
}

enum Seallll {
    WithOutCake,
    WithOutTea,
    WithOut,
}

enum NonCaps {
    Prefixçš„,
    PrefixTea,
    PrefixCake,
}

pub enum PubSeall {
    WithOutCake,
    WithOutTea,
    WithOut,
}

#[allow(pub_enum_variant_names)]
mod allowed {
    pub enum PubAllowed {
        SomeThis,
        SomeThat,
        SomeOtherWhat,
    }
}

// should not lint
enum Pat {
    Foo,
    Bar,
    Path,
}

// should not lint
enum N {
    Pos,
    Neg,
    Float,
}

fn main() {}
