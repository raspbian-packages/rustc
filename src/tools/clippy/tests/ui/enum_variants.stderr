error: Variant name ends with the enum's name
  --> $DIR/enum_variants.rs:14:5
   |
14 |     cFoo,
   |     ^^^^
   |
   = note: `-D enum-variant-names` implied by `-D warnings`

error: Variant name starts with the enum's name
  --> $DIR/enum_variants.rs:25:5
   |
25 |     FoodGood,
   |     ^^^^^^^^

error: Variant name starts with the enum's name
  --> $DIR/enum_variants.rs:26:5
   |
26 |     FoodMiddle,
   |     ^^^^^^^^^^

error: Variant name starts with the enum's name
  --> $DIR/enum_variants.rs:27:5
   |
27 |     FoodBad,
   |     ^^^^^^^

error: All variants have the same prefix: `Food`
  --> $DIR/enum_variants.rs:24:1
   |
24 | / enum Food {
25 | |     FoodGood,
26 | |     FoodMiddle,
27 | |     FoodBad,
28 | | }
   | |_^
   |
   = help: remove the prefixes and use full paths to the variants instead of glob imports

error: All variants have the same prefix: `CallType`
  --> $DIR/enum_variants.rs:34:1
   |
34 | / enum BadCallType {
35 | |     CallTypeCall,
36 | |     CallTypeCreate,
37 | |     CallTypeDestroy,
38 | | }
   | |_^
   |
   = help: remove the prefixes and use full paths to the variants instead of glob imports

error: All variants have the same prefix: `Constant`
  --> $DIR/enum_variants.rs:45:1
   |
45 | / enum Consts {
46 | |     ConstantInt,
47 | |     ConstantCake,
48 | |     ConstantLie,
49 | | }
   | |_^
   |
   = help: remove the prefixes and use full paths to the variants instead of glob imports

error: All variants have the same prefix: `With`
  --> $DIR/enum_variants.rs:78:1
   |
78 | / enum Seallll {
79 | |     WithOutCake,
80 | |     WithOutTea,
81 | |     WithOut,
82 | | }
   | |_^
   |
   = help: remove the prefixes and use full paths to the variants instead of glob imports

error: All variants have the same prefix: `Prefix`
  --> $DIR/enum_variants.rs:84:1
   |
84 | / enum NonCaps {
85 | |     Prefixçš„,
86 | |     PrefixTea,
87 | |     PrefixCake,
88 | | }
   | |_^
   |
   = help: remove the prefixes and use full paths to the variants instead of glob imports

error: All variants have the same prefix: `With`
  --> $DIR/enum_variants.rs:90:1
   |
90 | / pub enum PubSeall {
91 | |     WithOutCake,
92 | |     WithOutTea,
93 | |     WithOut,
94 | | }
   | |_^
   |
   = note: `-D pub-enum-variant-names` implied by `-D warnings`
   = help: remove the prefixes and use full paths to the variants instead of glob imports

error: aborting due to 10 previous errors

